### PR TITLE
isa, sim: added shifter_operand decoding function

### PR DESCRIPTION
**Pull request by parthmittal**
yah this compiles

### CODE REVIEW

File: include/sim.h [hunk 1]
```diff
@@ -7,6 +7,7 @@
 #define NB_REGS 16
 
 #define PC 15
+#define CFLAG 29
```
Comment (jailuthra) [inline include/sim.h]: rename this to CPSR_C

File: isa.c [hunk 2]
```diff
@@ -18,3 +18,161 @@ struct CPUState process_instruction(struct CPUState state)
...
+        LSLIMM = 0, //Logical shift left by immediate
+        LSLREG = 1, //Logical shift left by reg.
+        LSRIMM = 2, //Logical shift right by immediate
+        LSRREG = 3,
+        ASRIMM = 4, //Arithmetic shift right by immediate
+        ASRREG = 5, 
+        RORIMM = 6, //Rotate right by immediate
+        RORREG = 7
+    } s_type;
+    if (instruction & (1 << 25)) { //immediate
```
Comment (jailuthra) [inline isa.c]: don't use 25 directly, make a #define I_BIT 25

File: isa.c [hunk 3]
```diff
@@ -18,3 +18,161 @@ struct CPUState process_instruction(struct CPUState state)
...
+{
+    return (shiftee << (32 - shifter)) | (shiftee >> (shifter));
+}
+
+uint8_t get_bit(uint32_t from, uint8_t bitid)
+{
+    return (from >> bitid) & 1;
+}
+
+struct ShifterOperand * shifter_operand(struct CPUState state, uint32_t instruction)
```
Comment (jailuthra) [inline isa.c:150]: doxygen comments are recommended though not compulsory

**Review by jailuthra — COMMENTED** [review]
I haven't yet tested these. I think it's best to test these with asm files once we have everything ready

**Comment by jailuthra** [comment]
Rest looks cool and merge-able. Nice work.

File: isa.c [hunk 4]
```diff
@@ -18,3 +18,161 @@ struct CPUState process_instruction(struct CPUState state)
     // TODO: decode instruction @ PC and execute it
     return next_state;
 }
+
+uint32_t rotate_right(uint32_t shiftee, uint8_t shifter)
+{
+    return (shiftee << (32 - shifter)) | (shiftee >> (shifter));
```
Comment (jailuthra) [inline isa.c:142]: +100 street cred for this

**Comment by jailuthra** [comment]
also now that mrinal's pr has done a merge, resolve conflicts in isa.h in the next set of commits

**Comment by parthmittal** [comment]
@jailuthra bc

**Pull request merged**